new file:   .prettierrc.json

DubiumEkb opened the pull request:
modified:   public/index.html
	modified:   src/components/App/App.tsx
	modified:   src/components/BurgerConstructor/BurgerConstructor.tsx
	modified:   src/components/BurgerIngredients/BurgerIngredients.tsx
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/BurgerIngredientsItem.module.css
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/index.tsx
	new file:   src/components/IngredientDetails/IngredientDetails.module.css
	new file:   src/components/IngredientDetails/IngredientDetails.tsx
	new file:   src/components/Modal/Modal.module.css
	new file:   src/components/Modal/Modal.tsx
	new file:   src/components/ModalOverlay/ModalOverlay.module.css
	new file:   src/components/ModalOverlay/ModalOverlay.tsx
	new file:   src/components/OrderDetails/OrderDetails.module.css
	new file:   src/components/OrderDetails/OrderDetails.tsx
	modified:   src/index.tsx

DubiumEkb committed:
new file:   .prettierrc.json
	modified:   public/index.html
	modified:   src/components/App/App.tsx
	modified:   src/components/BurgerConstructor/BurgerConstructor.tsx
	modified:   src/components/BurgerIngredients/BurgerIngredients.tsx
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/BurgerIngredientsItem.module.css
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/index.tsx
	new file:   src/components/IngredientDetails/IngredientDetails.module.css
	new file:   src/components/IngredientDetails/IngredientDetails.tsx
	new file:   src/components/Modal/Modal.module.css
	new file:   src/components/Modal/Modal.tsx
	new file:   src/components/ModalOverlay/ModalOverlay.module.css
	new file:   src/components/ModalOverlay/ModalOverlay.tsx
	new file:   src/components/OrderDetails/OrderDetails.module.css
	new file:   src/components/OrderDetails/OrderDetails.tsx
	modified:   src/index.tsx

gennady-bars commented on src/components/App/App.tsx:
```diff
@@ -2,23 +2,50 @@
+		async function fetchData() {
+			fetch(url)
+				.then((response) => {
+					return response.json()
```
`					return response.json()
`

Любой ответ от сервера нужно проверять на `ok` и вызывать ошибку, если ответ не `ok`. Таким образом, в `catch` Вы будете ловить и ошибки сервера, и ошибки невалидных данных.


```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Ошибка ${res.status}`);
            })

```

gennady-bars commented on src/components/App/App.tsx:
```diff
@@ -2,23 +2,50 @@
+					setData(data.data)
+				})
+				.catch((error) => {
+					console.error(console.error())
```
`console.error(console.error())
`

нужно ошибку печатать, а не 2й раз вызывать `console.error()`

```js


				.catch((error) => {
					console.error(error)
				})
```

gennady-bars commented on src/components/App/App.tsx:
```diff
@@ -2,23 +2,50 @@
 import BurgerIngredients from "../BurgerIngredients/BurgerIngredients"
 import BurgerConstructor from "../BurgerConstructor/BurgerConstructor"
 
+const useFetch = (url: string) => {
```
Неплохой хук, но он только для `GET`-запросов и только для `useEffect`

его трудно будет переиспользовать в других местах, поэтому лучше сразу назвать `useFetchIngredients`, чтобы было понятно, что в нем получаете

gennady-bars replied:
Кастомные хуки принято помещать в папку `hooks` в свой файл, а потом экспортировать туда, где нужно

gennady-bars commented on src/components/BurgerIngredients/ui/BurgerIngredientsItem/index.tsx:
```diff
@@ -1,25 +1,49 @@
+			</button>
+
+			{show && (
+				<Modal title="Детали ингредиента" isOpen={show} onClose={handleClose} overlay={true}>
```
**Можно лучше**

*  лучше не делать в **каждом** ингредиенте по модальному окну. Нужно было сделать одно окно для деталей в ` BurgerIngredients` (или `App`) и передавать в каждый ингредиент функцию **клика**, в которой бы назначались данные для показа в окне

gennady-bars commented on src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,61 @@
+	const containerModal = document.getElementById("react-modals")
+
+	useEffect(() => {
+		const handleClose = (event: KeyboardEvent) => {
```
Отлично, что знаете про тип `KeyboardEvent` для событий клавиатуры

gennady-bars commented on src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,61 @@
+		}
+		window.addEventListener("keydown", handleClose)
+		return () => {
+			window.removeEventListener("keydown", handleClose)
```
правильно удаляется обработчик

gennady-bars commented on src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,61 @@
+	overlay?: boolean
+}
+
+const Modal = ({ children, title, isOpen, onClose, overlay = false }: Props) => {
```
`overlay = false ` по умолчанию нужно сделать `true`, ведь по умолчанию нужно оверлей всегда, а если потребуется, его можно отключать вручную через `false`


И тогда в коде проекта не нужно будет везде указывать `overlay={true}`

gennady-bars commented on src/components/Modal/Modal.tsx:
```diff
@@ -0,0 +1,61 @@
+				)}
+				{children}
+			</div>
+			{overlay === true && <ModalOverlay isOpen={isOpen} />}
```
* попапы не закрываются по оверлею


Нужно еще передавать функцию закрытия сюда

gennady-bars commented on src/components/App/App.tsx:
```diff
@@ -2,23 +2,50 @@
+				.then((data) => {
+					setData(data.data)
+				})
+				.catch((error) => {
```
Отлично, что ловите возможные ошибки в конце каждого запроса к серверу

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная

* Хорошая типизация компонентов
* Отлично, что ловите возможные ошибки в конце каждого запроса к серверу

но есть некоторые недочеты: 

* в консоли ошибки при открытии попапа заказа. Скрин http://joxi.ru/GrqaZ7gtz0Rnkr 
* попапы не закрываются по оверлею
* Скрин http://joxi.ru/MAj5QOltdB1RL2 видна граница оверлея и попап уходит за пределы экрана. Модальное окно должно быть всегда ровно посередине экрана. Для этого нужно указать ему и оверлею в `css` позицию `fixed`.
* Любой ответ от сервера нужно проверять на `ok` и вызывать ошибку, если ответ не `ok`. Таким образом, в `catch` Вы будете ловить и ошибки сервера, и ошибки невалидных данных.
* Кастомные хуки принято помещать в папку `hooks` в свой файл, а потом экспортировать туда, где нужно



**Можно лучше**

*  лучше не делать в **каждом** ингредиенте по модальному окну. Нужно было сделать одно окно для деталей в ` BurgerIngredients` (или `App`) и передавать в каждый ингредиент функцию **клика**, в которой бы назначались данные для показа в окне



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

DubiumEkb committed:
modified:   src/components/App/App.tsx
	modified:   src/components/BurgerIngredients/BurgerIngredients.module.css
	modified:   src/components/BurgerIngredients/BurgerIngredients.tsx
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/BurgerIngredientsItem.module.css
	modified:   src/components/BurgerIngredients/ui/BurgerIngredientsItem/index.tsx
	modified:   src/components/Modal/Modal.module.css
	modified:   src/components/Modal/Modal.tsx
	modified:   src/components/ModalOverlay/ModalOverlay.module.css
	modified:   src/components/ModalOverlay/ModalOverlay.tsx
	modified:   src/components/OrderDetails/OrderDetails.tsx
	new file:   src/utils/hooks/useFetch.ts

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
